### PR TITLE
Draft of summary for HW FP Atomics and fine/coarse grained memory

### PR DESCRIPTION
**Pull request by afanfa**
Hi all,
this is an initial draft for the summary on HW FP atomics and fine/coarse grained memory. A link to the more detailed version of the document is currently missing. The plan is to have a PDF linked at the bottom but I am open to other ideas.

@nicholasmalaya, @abbots, @arghdos

### CODE REVIEW

**Comment by afanfa** [comment]
@tom-papatheodore I have just made a few adjustments to this PR.
Could you or somebody else review it?
Thanks.

File: systems/crusher_quick_start_guide.rst [hunk 1]
```diff
@@ -1444,6 +1444,64 @@ The above log messages indicate the type of image required by each device, given
...
+
+**Fine grained** memory allows CPUs and GPUs to synchronize (via atomics) and coherently communicate with each other while the GPU kernel is running, allowing more advanced programming patterns. The additional visibility impacts the performance of fine grained allocated memory.
+The fast hardware-based Floating point (FP) atomic operations available on MI250X are assumed to be working on coarse grained memory regions; when these instructions are applied to a fine-grained memory region, they will silently produce a no-op. To avoid returning incorrect results, the compiler never emits hardware-based FP atomics instructions by default, even when applied to coarse grained memory regions. Currently, users can use the `-munsafe-fp-atomics` flag to suggest to the compiler that hardware-based FP atomics should be emitted.
```
Comment (tom-papatheodore) [inline systems/crusher_quick_start_guide.rst:1459]: Another space is needed here to format correctly. Currently, the paragraph starting on 1459 is just added to the "Fine grained" sub-section, which isn't intended.

File: systems/crusher_quick_start_guide.rst [hunk 2]
```diff
@@ -1444,6 +1444,64 @@ The above log messages indicate the type of image required by each device, given
...
+Using hardware-based FP atomics translates in a substantial performance improvement over the default choice.
+
+Users applying floating point atomic operations (e.g., atomicAdd) on memory regions allocated via regular hipMalloc() can safely apply the `-munsafe-fp-atomics` flags to their codes to get the best possible performance and leverage hardware supported floating point atomics.
+Atomic operations supported in hardware on non-FP datatypes  (e.g., INT32) will work correctly regardless of the nature of the memory region used.
+
+In ROCm-5.1 and earlier versions, the flag `-munsafe-fp-atomics` is interpreted as a suggestion by the compiler, whereas from ROCm-5.2 the flag will always enforce the use of fast hardware-based FP atomics.
```
Comment (tom-papatheodore) [inline systems/crusher_quick_start_guide.rst:1465]: In the last sentence in the paragraph starting on 1459, it says "to suggest to the compiler", but here in this sentence it says this is only true for `rocm/5.1.0` and earlier. Should we change the previous (more general) sentence above to read "to enforce the compiler to"?

File: systems/crusher_quick_start_guide.rst [hunk 3]
```diff
@@ -1444,6 +1444,64 @@ The above log messages indicate the type of image required by each device, given
...
+
+The following tables summarize the result granularity of various combinations of allocators, flags and arguments.
+
+For hipHostMalloc, the following table shows the nature of the memory returned based on the flag passed as argument.
+
++---------------------------+-----------------+
```
Comment (tom-papatheodore) [inline systems/crusher_quick_start_guide.rst:1471]: I would probably add a column in this table for the API call - just to be consistent with the other 2 tables.

**Review by tom-papatheodore — COMMENTED** [review]
@afanfa I've added some comments to your PR. In addition to those comments, I see there are some inconsistencies in the text when you refer to a API call or (more generally) when you refer to code; sometimes it's just normal text, other times it's in italics. I would suggest using double backticks around inline code to use code formatting. So something like ``hipMalloc``. Otherwise, this looks great and will be a helpful improvement for the Crusher users.

File: systems/crusher_quick_start_guide.rst (hunk 2 above)
Comment (afanfa) [reply systems/crusher_quick_start_guide.rst:1465]: yes, that makes sense.

**Review by afanfa — COMMENTED** [review]
I have introduced all the recommended changes. Thanks

**Review by tom-papatheodore — APPROVED** [review]
LGTM! Thanks @afanfa!

**Pull request merged**